### DISCUSSION TITLE
My database has lots of songs only with Overlaps / Cross Start Next (no Cue Out or Fade Out defined). How can i calculate fast Cue Out Points?

### DISCUSSION THREAD

**Discussion by unudelaradio** [AzuraCast/AzuraCast#7578] [Q&A] [answered]
Hello!

In all these years using AzuraCast, i created a strong and large Custom Cue Points database for my songs. In the past, with older versions of AzuraCast there was enough to set only an `Overlap` in Visual Cue Editor for the desired `Start Mix Point` to work fine. So now, lots of my songs remain with only these `Overlaps (Start Next)` defined.

Is there a method to calculate in bulk `Cue Out` for all these songs? Then, after i get the `Cue Out` Points, i will calculate the `Fade Out` Points based on a `Cue Out minus Overlap` formula.

I want to enable `AutoCue`, but since i have lots of songs with only a Overlap / Start Next defined, there are two solutions: either i clean all these overlaps (which i don't want) either i should define `Fade Out` points, so that all the mix points to be respected and work fine.

I was wondering if there's a way to get the exact length of these songs as a Cue Out? In decimals.

I remember when i updated from v0.19.7 to v0.20.0, all Overlaps where recalculated. A 2.9 `Overlap Duration` became a 221.8 `Start Next` (something like that). At that time it was great that `Cue Out` fields to be autopopulated too. So in all the cases where there's no `Cue Out` defined, but a `Start Next` is present.... the `Cue Out` field to be automatically calculated/populated in Advanced Edit Media, as the total length of the song.

The ideal scenario was that `Start Next` to be respected no matter if `Fade Out` or `Cue Out` are defined.

**Comment by BusterNeece** [comment]
If you only have some of the AutoCue fields, we automatically fill in the cue-in/out and fade-in/out with sensible default values, i.e. the cue-out value is the duration of the track and the fade-in/out are set to the global crossfade settings.

**Reply by unudelaradio** [reply]
I would find very useful for me a new column called `Length` (duration of the track) in the Bulk Media Import/Export CSV. So i can copy some `Length` values and paste it to `Cue Out` column for specific songs (the ones with empty Cue Out points). That would solve my issue.

**Reply by BusterNeece** [reply, marked as answer]
@unudelaradio A solid point, and very easy to implement. Done in Rolling!

**Reply by unudelaradio** [reply]
Wow! That's awesome! Lots of thanks!
